libcontainer/nsenter/nsexec.c

```diff
@@ -131,20 +140,40 @@ int setns(int fd, int nstype)
 }
 #endif
 
+void write_log_with_info(const char *level, const char *function, int line, const char *format, ...)
+{
+	static char message[1024];
+	va_list args;
+
+	if (logfd < 0 || level == NULL)
+		return;
+
+	va_start(args, format);
+	if (vsnprintf(message, 1024, format, args) < 0)
+		return;
+	va_end(args);
+
+	if (dprintf(logfd, "{\"level\":\"%s\", \"msg\": \"%s:%d %s\"}\n", level, function, line, message) < 0)
+		return;
+}
+
+#define write_log(level, fmt, ...) \
+	write_log_with_info(level, __FUNCTION__, __LINE__, fmt, ##__VA_ARGS__)
```

Comment:
C-foo: You should (just to be safe) parenthesise all of the forwarded arguments, like so:

```
write_log_with_info((level), __FUNCTION__, __LINE__, (fmt), ##__VA_ARGS__)
```

This is because C macros aren't hygenic and you can end up in very odd situations where the expansion is counter-intuitive.

Reply:
Done